src/js/components/settings/organization.js

```diff
@@ -165,55 +150,33 @@ export class MyOrganization extends React.Component {
-            {this.state.cancelSubscription && (
-              <div className="margin-left-small margin-right-small">
-                <p className="margin-top-large">
-                  The request won&#39;t be fulfilled immediately.
-                  <br />
-                  Our Support Team will act upon it and be back to you soon.
+            {this.state.showCancelRequestConfirmation ? (
+              <Alert className="margin-top-large" severity="error" style={{ maxWidth: '500px' }}>
+                <p>We&#39;ve started the process to cancel your plan and deactivate your account.</p>
+                <p>
+                  We&#39;ll send you an email confirming your deactivation. If you have any question at all, contact us at{' '}
+                  <strong>
+                    <a href="mailto:support@mender.io">support@mender.io</a>
+                  </strong>
                 </p>
-                <p>Please select the reason for your cancellation to help us improve our service:</p>
-                <div className="flexbox space-between">
-                  <FormControl id="reason-form">
-                    <InputLabel id="reason-selection-label">Reason</InputLabel>
-                    <Select
-                      labelId="reason-selection-label"
-                      id={`reason-selector-${this.cancelSubscriptionReasons}.length}`}
-                      value={this.state.cancelSubscriptionReason}
-                      validations="isLength:1"
-                      onChange={evt => this._setCancelSubscriptionReason(evt)}
-                      required={true}
-                    >
-                      {this.cancelSubscriptionReasons.map(item => (
-                        <MenuItem key={item.value} value={item.value}>
-                          {item.label}
-                        </MenuItem>
-                      ))}
-                    </Select>
-                  </FormControl>
-                  <Button
-                    variant="contained"
-                    color="secondary"
-                    onClick={() => self.setState({ showCancelRequestConfirmationDialog: true })}
-                    disabled={this.state.cancelSubscriptionReason == ''}
-                  >
-                    Submit the request
-                  </Button>
-                </div>
-              </div>
+              </Alert>
+            ) : (
+              <p className="margin-left-small margin-right-small">
+                <a href="" onClick={e => self.handleCancelSubscription(e)}>
+                  Cancel subscription and deactivate account
+                </a>
+              </p>
             )}
+            <CancelRequestConfirmationDialog
+              open={this.state.cancelSubscriptionConfirmation}
+              onCancel={() => self.setState({ cancelSubscriptionConfirmation: false })}
+              onSubmit={() => this._cancelSubscriptionConfirmationSubmit()}
+            />
+            <CancelRequestDialog
```

Comment:
the dialog remains stale if you open it, choose a reason to cancel, cancel and reopen it - you probably could get around this if you render it conditionally so `{this.state.cancelSubscription && <CancelRequestDialog ... />}` as it will be recreated entirely (as in the constructor is executed each time) once `cancelSubscription` is truthy...
this way you can then omit the open prop, since the dialog would - if it gets invoked - always be open

Reply:
Fixed